cmd/kubeadm/app/phases/addons/dns/dns_test.go

```diff
@@ -550,3 +553,633 @@ func createClientAndCoreDNSManifest(t *testing.T, corefile, coreDNSVersion strin
 	}
 	return client
 }
+
+func Test_deployedDNSReplicas(t *testing.T) {
+	type args struct {
+		dnsreplicas int32
+		replicas    int32
+	}
+	tests := []struct {
+		name           string
+		args           args
+		deploymentsize int
+		want           int32
+		wantErr        bool
+	}{
+		{
+			name: "one coredns addon deployment",
+			args: args{
+				dnsreplicas: 2,
+				replicas:    5,
+			},
+			deploymentsize: 1,
+			want:           2,
+			wantErr:        false,
+		},
+		{
+			name: "no coredns addon deployment",
+			args: args{
+				dnsreplicas: 2,
+				replicas:    5,
+			},
+			deploymentsize: 0,
+			want:           5,
+			wantErr:        false,
+		},
+		{
+			name: "multiple coredns addon deployments",
+			args: args{
+				dnsreplicas: 2,
+				replicas:    5,
+			},
+			deploymentsize: 3,
+			want:           5,
+			wantErr:        true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := newMockClientForTest(t, tt.args.dnsreplicas, tt.deploymentsize)
+			got, err := deployedDNSReplicas(client, tt.args.replicas)
```

Comment:
i don't think we need to test variadic number of DNS replicas?
see the code around - deployedDNSReplicas() we always call it with 2, unless i'm mistaken.